robosuite/scripts/collect_human_demonstrations.py

```diff
@@ -15,11 +16,95 @@
 import numpy as np
 
 import robosuite as suite
+import robosuite.utils.transform_utils as T
 from robosuite.controllers import load_composite_controller_config
 from robosuite.controllers.composite.composite_controller import WholeBody
+from robosuite.controllers.parts.arm import InverseKinematicsController, OperationalSpaceController
+from robosuite.utils.log_utils import ROBOSUITE_DEFAULT_LOGGER
 from robosuite.wrappers import DataCollectionWrapper, VisualizationWrapper
 
 
+def get_arm_ref(env, robot, arm):
+    """
+    Extracts the reference pose of the specified arm of the robot.
+    """
+    if robot.composite_controller_config["type"] in ["WHOLE_BODY_MINK_IK", "HYBRID_WHOLE_BODY_MINK_IK"]:
+        ref_frame = robot.composite_controller.composite_controller_specific_config.get("ik_input_ref_frame", "world")
+
+        # general case
+        id = robot.robot_model.idn
+        site_name = f"gripper{id}_{arm}_grip_site"
+        # update next target based on current achieved pose
+        pos = env.sim.data.get_site_xpos(site_name).copy()
+        ori = env.sim.data.get_site_xmat(site_name).copy()
+        # convert target in world coordinate to
+        pose_in_world = np.eye(4)
+        pose_in_world[:3, 3] = pos
+        pose_in_world[:3, :3] = ori
+        pose_in_base = robot.composite_controller.joint_action_policy.transform_pose(
+            src_frame_pose=pose_in_world,
+            src_frame="world",  # mocap pose is world coordinates
+            dst_frame=ref_frame,
+        )
+        pos, ori = pose_in_base[:3, 3], pose_in_base[:3, :3]
+        axisangle = T.quat2axisangle(T.mat2quat(ori))
+
+        abs_action = np.concatenate([pos, axisangle])
+
+        return abs_action
+    elif robot.composite_controller_config["type"] in ["WHOLE_BODY_IK"]:
+        id = robot.robot_model.idn
+        site_name = f"gripper{id}_{arm}_grip_site"
+        # update next target based on current achieved pose
+        pos = env.sim.data.get_site_xpos(site_name).copy()
+        ori = env.sim.data.get_site_xmat(site_name).copy()
+
+        axisangle = T.quat2axisangle(T.mat2quat(ori))
+
+        abs_action = np.concatenate([pos, axisangle])
+
+        return abs_action
+    else:
+        raise NotImplementedError
+
+
+def get_abs_arm_static_target(robot, arm, env):
```

Comment:
why's the fn called `get_abs_arm_static_target`? what's an absolute arm? does it mean arm's controller is absolute and doesn't work for relative action space? Can the function just be to `get_arm_target()` or `get_arm_current_target`?